### ISSUE TITLE
Documentation / what mean Loss, transfer learning / object detection

### ISSUE THREAD

**Issue by mrsaw12** [apple/turicreate#2218]
Hi Guys,
i's not really a issue, but i dont know where i can ask this, because it's too turi create specific. I guess the answers help too other people.

i'm train at this moment a object detection model, and have still following questions:

1) what means LOSS / what is LOSS? 
+--------------+--------------+--------------+
| Iteration    | Loss         | Elapsed Time |
+--------------+--------------+--------------+
| 1            | 9.083        | 12.0         |
| 7            | 9.027        | 22.8         |
....
| 47509        | 1.432        | 82039.8      |
| 47515        | 1.484        | 82050.4      |
| 47522        | 1.437        | 82062.1      |
i understand, if Loss is smaller, than better. But it's possible that Loss can get to "small" and this shows maybe an overfitting? What i'm asking for, is a LOSS of 1.0 the best value that i can get and a value that i want to have?

2) in the docs i read, that **image classification** has **transfer learning**, _have object detection this too_?  In the docs i found the same "model" option, but at image classification the docs definitely write, that it's "transfer learning", but at object detection not. It's not clear what happen if i set **model="darknet-yolo"**, because if i not set it, it's still a darknet-yolo model...

3) i read in the docs / user guide, that 30 examples (bounding boxes) are enough... but for better quality i need like 200... in my case i train the model to recognize from a TOP view "persons", because the ready to download models cant to this / or have a bad performance. I have round about 800 pictures with 30.000 bounding boxes, but most of them have the same flooring. So it's better to have more pictures or maybe just 200 bounding boxes are already enough? And if the flooring often same, it's better to change it by photoshop with other flooring or not to train to much with this flooring the model?
Detailed the questions are:

- Given a maximum size of pictures / bounding boxes? or is more ever better?
- For example if i have persons behind desk / chair / or other persons, it's better to annotate only what i can see from a person, or too whats on the front, and where i think the foots are?
![no_foot](https://user-images.githubusercontent.com/8724611/62724029-5c2ad780-ba12-11e9-9432-da26b86796f3.png) ![with_foot](https://user-images.githubusercontent.com/8724611/62724024-57662380-ba12-11e9-9638-f5fa24b0395d.png)
- it's maybe better to train the model not only to understand how persons look, too to understand what a chair / desk / backpack / bag and so one is, that it can better recognize a person?

4) If i understand it right, than transfer learning transfer the layers, that a other model already trained to my new model. So, it should be able to recognize for example chairs, desk and so one, and i dont have to train this again? Or not? But if i train for example the class person, than i just have only one "class" in my model. So what i did wrong here? 

5) Can you suggest a good guide / book that maybe explain more the things of data preparation and transfer learning?

Thank you very much.

**Comment by nickjong** [comment]
Let me take a pass at some of your questions!

1. The loss reported in the progress table is the "YOLO" loss defined in the model/algorithm described in our user guide here: https://apple.github.io/turicreate/docs/userguide/object_detection/how-it-works.html It's essentially a measure of the distance/discrepancy between what the model is currently predicting and the labeled ground truth. The smallest possible loss is 0, corresponding to predictions that exactly match the labeled data. Overfitting is certainly a potential issue. Right now, what you can do is try running the model for different numbers of iterations and comparing the resulting validation metrics on a held-out set of data. One thing we're thinking about now is how to make it easier to reason about checkpointing / early stopping to streamline this process.

**Comment by mrsaw12** [comment]
Hi nickjing, thanks for fast response. and yes your answers helped me already much. Following 2 more questions, would be much appreciated if you can answer to it too :)

3) I guess your answer for no. 4 was more related to my question no. 3 of correct "Data preparation"
The user guide described how create from dataset "IG02" annotations. Data IG02 is rich annotated, because it has masks and occlusions. So my questions is: _understand turicreate occlusion?_ If i want that the model recognize a "full" person (with legs, that are behind something) so how i have annotate this? First the Person and where i think the legs/foots end and than a box over the part of the desk/bar and give them "no person" label? 
But what if i have 2-3 persons, and one is behind the other one. First person is 100% visible, and the person behind only for 50% visible. Is in this case following annotation the correct one?
![person_behind_person](https://user-images.githubusercontent.com/8724611/62735554-bab07f80-ba2b-11e9-9c0e-d06abb37614c.png)


4) sorry i mixed up here something between "transfer learning" and "weights", my questions was more like, "it's possible to download yolo pretrained weights and integrate it to my model with my own data." (so that i don't need train again the model for "chairs, desks, cups and so one". and just can add my trained class to the already trained classes?)

Thank you very much for your time.

**Comment by nickjong** [comment]
Sorry for the delay in responding!

1. At present, I'm afraid the best way to guard against overfitting is to evaluate the trained model, for different numbers of iterations, against a held-out test set. We're in the process of overhauling our Object Detection implementation, and simplifying this task is something we're working on. Perhaps by just periodically giving the validation performance on a validation set, during training.
2. I cannot say with confidence how well the model will handle the occluded examples you gave, but honestly I think your three annotations are essentially what I would use. 
3. Our implementation already downloads a pretrained model (darknet) and uses that as the starting point for fine-tuning a model that regresses against your labeled data. We don't currently have a way to take, say, a detector model trained for chairs, desks, and cups and adding support for a new class while preserving the existing behavior on the old classes (without having training examples of chairs, desks, and cups to guide the tuning of the weights).
